src/renderers/shared/fiber/ReactFiberUpdateQueue.js

```diff
@@ -469,19 +465,14 @@ function beginUpdateQueue(
 }
 exports.beginUpdateQueue = beginUpdateQueue;
 
-function commitCallbacks(finishedWork : Fiber, callbackList : UpdateQueue, context : mixed) {
-  const stopAfter = callbackList.last;
-  let update = callbackList.first;
-  while (update) {
-    const callback = update.callback;
-    if (typeof callback === 'function') {
-      callback.call(context);
-    }
-    if (update === stopAfter) {
-      break;
-    }
-    update = update.next;
+function commitCallbacks(finishedWork : Fiber, queue : UpdateQueue, context : mixed) {
+  const callbackList = queue.callbackList;
+  if (!callbackList) {
+    return;
+  }
+  for (let i = 0; i < callbackList.length; i++) {
+    const callback = callbackList[i];
+    callback.call(context);
   }
-  finishedWork.callbackList = null;
```

Comment:
This reminds me though that in `cloneUpdateQueue`, some of those fields should just be reset since by the time we clone from current, they're no longer valid. Not an observable change but it's what `cloneFiber` does so better to be consistent.